packages/app/components/settings/settings-account-slot.tsx

```diff
@@ -56,15 +65,55 @@ export const SettingAccountSlotFooter = () => {
         </Text>
         <View tw="h-4" />
         <View tw="flex flex-row">
-          <Link href="mailto:support@tryshowtime.com">
-            <Button variant="danger" size="small" onPress={handleDeleteAccount}>
-              <Text>Delete Account</Text>
-            </Button>
-          </Link>
+          <Button
+            variant="danger"
+            size="small"
+            onPress={() => {
+              setShowDeleteConfirmation(true);
+              setError("");
+            }}
+          >
+            <Text>Delete Account</Text>
+          </Button>
         </View>
         <View tw="h-4" />
         <ClearCacheBtn />
       </View>
+      <ModalSheet
+        title="Delete Account?"
+        visible={showDeleteConfirmation}
+        close={() => setShowDeleteConfirmation(false)}
+        onClose={() => setShowDeleteConfirmation(false)}
+      >
+        <View>
+          <View tw="mb-4">
+            <Text tw="text-base text-gray-900 dark:text-gray-100">
+              Are you sure you want to delete your account? This action cannot
+              be undone
+            </Text>
+            <Text tw="py-4 text-base text-gray-900 dark:text-gray-100">
+              Type your <Text tw="font-bold">username</Text> to confirm.
+            </Text>
+          </View>
+          <Fieldset
+            onChangeText={(t) => (username.current = t)}
+            placeholder="@username"
```

Comment:
if the placeholder is real `username` will be better, just like GitHub remove the repo, wdyt?

Reply:
i kept it like that to make it harder. But we can do it the Github way too

Reply:
🤣 that makes sense!

Reply:
updated